perl/shop/t/Basket.t

```diff
@@ -1,319 +1,364 @@
+        'Quantity' => 50,
+    };
+    $BasketService->addProductLineItem( $BasketPath, [$ProductLineItem] )->result;
 
-    return $hCreate->{'Path'};
-}
+    my $hBasketInfo = _getInfo($BasketPath);
 
-sub testGetInfoReference {
-    my ( $basketPath, $basketRef ) = @_;
+    my $LineItemAlias = $hBasketInfo->{LineItemContainer}->{'ProductLineItems'}->[0]->{'Alias'};
 
-    my $ahResults = $BasketService->getInfo( [$basketPath], [], [] )->result;
-    is( scalar @$ahResults, 1, 'getInfo result set' );
+    my $ahResultDelete = $BasketService->deleteLineItem( $BasketPath, [$LineItemAlias] )->result;
+    is( scalar @$ahResultDelete, 1, 'testDeleteLineItem: result count' );
+    my $hResultDelete = $ahResultDelete->[0];
+    ok( !$hResultDelete->{'Error'}, 'testDeleteLineItem: check error' );
+    diag $hResultDelete->{'Error'}->{'Message'} . "\n" if $hResultDelete->{'Error'};
 
-    my $hInfo = $ahResults->[0];
-    ok( !$hInfo->{'Error'}, 'getInfo: no error' );
-    diag "Error: $hInfo->{'Error'}\n" if $hInfo->{'Error'};
+    my $LineItemPath = "$BasketPath/LineItemContainer/$LineItemAlias";
+    is( $hResultDelete->{'Path'},    $LineItemPath, 'testDeleteLineItem: line item path' );
+    is( $hResultDelete->{'deleted'}, 1,             'testDeleteLineItem: deleted' );
+}
 
-    my $hLineItemContainer  = $hInfo->{'LineItemContainer'};
-    my $hLineItemContainer2 = $basketRef->{'LineItemContainer'};
-    ok( $hLineItemContainer->{'TaxArea'}    eq $hLineItemContainer2->{'TaxArea'},       'tax area' );
-    ok( $hLineItemContainer->{'TaxModel'}   eq $hLineItemContainer2->{'TaxModel'},      'tax model' );
-    ok( $hLineItemContainer->{'CurrencyID'} eq $hLineItemContainer2->{'CurrencyID'},    'currencyid' );
+sub testExistsBasket {
+    my $BasketPath = _setupTestBasketInDB();
 
-    ok( $hLineItemContainer->{'ProductLineItems'}, 'list of all product line items');
-    my @ProductLineItems =
-        map {{$_->{'Product'}, { 'Quantity' => $_->{'Quantity'}, 'OrderUnit' => $_->{'OrderUnit'}}}} sort {$a->{Product} le $b->{Product}}
-        @{$hLineItemContainer->{'ProductLineItems'}};
-    my @ProductLineItemsRef =
-        map {{$_->{'Product'},{ 'Quantity' => $_->{'Quantity'}, 'OrderUnit' => $_->{'OrderUnit'}}}} sort {$a->{Product} le $b->{Product}}
-        @{$hLineItemContainer2->{'ProductLineItems'}};
+    my $ahResultExists = $BasketService->exists( [$BasketPath] )->result;
+    is( scalar @$ahResultExists, 1, 'testExistsBasket: response format' );
 
-    is_deeply( \@ProductLineItems, \@ProductLineItemsRef, 'product line items');
+    my $hResultExists = $ahResultExists->[0];
+    ok( !$hResultExists->{'Error'}, 'testExistsBasket: check error' );
+    diag $hResultExists->{'Error'}->{'Message'} . "\n" if $hResultExists->{'Error'};
 
-    return $hInfo;
+    is( $hResultExists->{'Path'},   $BasketPath, 'testExistsBasket: basket path' );
+    is( $hResultExists->{'exists'}, 1,           'testExistsBasket: exists' );
 }
 
 sub testDeleteBasket {
```

Comment:
Test checks only the response of the basket service. An additional call to `exists` could verify that the basket has really been deleted.